### PR TITLE
Payment Entry - Latest Exchange Rate not fetched

### LINKED ISSUE #8919: Journal Entry not fetching currency exchange rate set in Currency Exchange List master
ERPNext: v8.0.24 (master)

Frappe Framework: v8.0.42 (master)

-------------------------------------------------------

Steps to replicate

1) Set Exchange Rate
For example, this is set on 31 January 2017

image

image

2) Create Journal Entry

3) Select Posting Date as 31 January 2017

4) Row 1 Journal Entry Detail - Select a Foreign Currency Account. Click to view detail.
The exchange rate isn't fetched. It is still 1. As such the amount in Company Currency is wrong.

image

Expected Behaviour:
Exchange rate to fetch the latest from Posting Date. This will correctly calculate amount in company currency.

### PR DESCRIPTION
**Pull request by tundebabzy**
This fixes #8919, #9180

These two issues arise from the fact that "Pay" payment entries retrieve the average exchange rate used for transactions to the specified bank account. The side effect of this is that ERPNext can present different exchange rates for two different accounts with the same currency (#8919). Also, ERPNext will not be able to provide any exchange rate if the bank account balance is zero (#9180).

Technically, [IAS 21](https://www.iasplus.com/en/standards/ias/ias21#link8) does not allow for averaging exchange rates. Rather each transaction should be converted at the closing exchange rate on the day the transaction took place. Thus, the exchange rate automatically drawn should be the last saved exchange rate.

### CODE REVIEW

**Comment by tundebabzy** [comment]
Here's the difference

### Before
peek 2017-06-08 16-20

### After
peek 2017-06-08 16-35

**Review by nabinhait — CHANGES_REQUESTED** [review]
The same issue is there in Journal Entry when bank account is credited, it takes average there as well.

File: erpnext/accounts/doctype/journal_entry/journal_entry.py [hunk 1]
```diff
@@ -871,14 +871,9 @@ def get_exchange_rate(posting_date, account=None, account_currency=None, company
...
 
-		elif account_details and account_details.account_type == "Bank" and \
-			((account_details.root_type == "Asset" and flt(credit) > 0) or
-				(account_details.root_type == "Liability" and debit)):
-			exchange_rate = get_average_exchange_rate(account)
-
 		# The date used to retreive the exchange rate here is the date passed
 		# in as an argument to this function.
-		if not exchange_rate and account_currency and posting_date:
+		elif account_currency and posting_date:
```
Comment (nabinhait) [inline erpnext/accounts/doctype/journal_entry/journal_entry.py]: Why removed `not exchange_rate`? User will not be able to set exchange_rate manually.

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/accounts/doctype/journal_entry/journal_entry.py (hunk 1 above)
Comment (tundebabzy) [inline erpnext/accounts/doctype/journal_entry/journal_entry.py]: i'll ammend the commit

**Comment by addymuliady** [comment]
Greetings @nabinhait 

Is it possible to merge this into hotfix as it is critical to my accounting. I cannot overwrite the average exchange rate and am stuck with my work until this fix is merged to master.

For your kind perusal please. Thanks!

**Pull request merged**